src/executor/test/ConjunctPathTest.cpp

```diff
@@ -758,5 +1232,194 @@ TEST_F(ConjunctPathTest, AllPathsFourStepsPath) {
+            row.values.emplace_back(4);
+            expected.rows.emplace_back(std::move(row));
+        }
+    }
+    {
+        // 2->6->7->8->9
+        Row row;
+        Path path;
+        path.src = Vertex("2", {});
+        path.steps.emplace_back(Step(Vertex("6", {}), 1, "edge1", 0, {}));
+        path.steps.emplace_back(Step(Vertex("7", {}), 1, "edge1", 0, {}));
+        path.steps.emplace_back(Step(Vertex("8", {}), 1, "edge1", 0, {}));
+        path.steps.emplace_back(Step(Vertex("9", {}), 1, "edge1", 0, {}));
+        row.values.emplace_back(std::move(path));
+        row.values.emplace_back(4);
+        expected.rows.emplace_back(std::move(row));
+    }
+    {
+        // 3->4->7->8->9
+        Row row;
+        Path path;
+        path.src = Vertex("3", {});
+        path.steps.emplace_back(Step(Vertex("4", {}), 1, "edge1", 0, {}));
+        path.steps.emplace_back(Step(Vertex("7", {}), 1, "edge1", 0, {}));
+        path.steps.emplace_back(Step(Vertex("8", {}), 1, "edge1", 0, {}));
+        path.steps.emplace_back(Step(Vertex("9", {}), 1, "edge1", 0, {}));
+        row.values.emplace_back(std::move(path));
+        row.values.emplace_back(4);
+        expected.rows.emplace_back(std::move(row));
+    }
+    std::sort(expected.rows.begin(), expected.rows.end(), comparePath);
+    auto resultDs = result.value().getDataSet();
+    std::sort(resultDs.rows.begin(), resultDs.rows.end(), comparePath);
+    EXPECT_EQ(resultDs, expected);
+    EXPECT_EQ(result.state(), Result::State::kSuccess);
+}
+
+TEST_F(ConjunctPathTest, multiplePairThreeSteps) {
+    auto* conjunct = ConjunctPath::make(qctx_.get(),
+                                        StartNode::make(qctx_.get()),
+                                        StartNode::make(qctx_.get()),
+                                        ConjunctPath::PathKind::kFloyd,
+                                        5);
+    conjunct->setLeftVar("forwardPath3");
+    conjunct->setRightVar("backwardPath3");
+    conjunct->setColNames({"_path", "cost"});
+    auto conjunctExe = std::make_unique<ConjunctPathExecutor>(conjunct, qctx_.get());
+    auto future = conjunctExe->execute();
+    auto status = std::move(future).get();
+    EXPECT_TRUE(status.ok());
+    auto& result = qctx_->ectx()->getResult(conjunct->outputVar());
+
+    DataSet expected;
+    expected.colNames = {"_path", "cost"};
+    {
+        // 1->5->7->10->11->12, 1->6->7->10->11->12
+        for (auto i = 5; i < 7; i++) {
+            Row row;
+            Path path;
+            path.src = Vertex("1", {});
+            path.steps.emplace_back(Step(Vertex(folly::to<std::string>(i), {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("7", {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("10", {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("11", {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("12", {}), 1, "edge1", 0, {}));
```

Comment:
Why not to provide a path build function to extract this logics for all tests, like:

```cpp
Path createPath(const string &src, const std::vector<string>& steps) {
  Path path;
  path.src = src;
  for (auto& step : steps) {
    path.steps.emplace_back(Step(Vertex(step, {)), 1, "edge1", 0, {});
  }
  return path;
}

// usage
auto path = createPath("1", {folly::to<string>(i), "7", "10", "11"});
```